packages/svelte/src/compiler/phases/3-transform/utils.js

```diff
@@ -287,10 +287,11 @@ export function clean_nodes(
 					!first.attributes.some(
 						(attribute) => attribute.type === 'Attribute' && attribute.name.startsWith('--')
 					))),
-		/** if a component or snippet starts with text, we need to add an anchor comment so that its text node doesn't get fused with its surroundings */
+		/** if a component/snippet/each block starts with text, we need to add an anchor comment so that its text node doesn't get fused with its surroundings */
 		is_text_first:
 			(parent.type === 'Fragment' ||
 				parent.type === 'SnippetBlock' ||
+				parent.type === 'EachBlock' ||
```

Comment:
That would break the case where you have only text nodes:
```svelte
{#each 'abc' as l}
	{l}
{/each}
```
It doesn't result in a hydration mismatch, likely because it's repaired, but from debugging I can clearly see it wrongly fuses the text nodes together and hydration does destroy and recreated nodes.